torch/distributed/_composable/fsdp/fully_shard.py

```diff
@@ -0,0 +1,46 @@
+from typing import Any
+
+import typing_extensions
+
+import torch.nn as nn
+
+from torch.distributed._composable import contract
+from torch.distributed._composable_state import _insert_module_state
+
+
+@contract()
+def fully_shard(
+    module: nn.Module,
+):
+    if isinstance(module, (nn.ModuleList, nn.ModuleDict)):
+        raise ValueError(
+            f"fully_shard does not support containers that do not implement forward: {module}"
+        )
+    state = fully_shard.state(module)
+    _insert_module_state(module, state)
+    # Place FSDP leftmost for highest priority in the method resolution order
+    cls = module.__class__
+    dct = {"__deepcopy__": unimplemented_deepcopy}
+    new_cls = type(f"FSDP{cls.__name__}", (FSDP, cls), dct)
```

Comment:
I thought this should be the below instead?
```
f"FSDP<{cls.__name__}>"
```

Reply:
For my understanding, is this mainly a stylistic difference of what the new class name is? Or, is there another reason why we want the `<>`?

If we have the `<>`, is that a valid class name? I was not sure how you can do `isinstance(module, FSDP<Linear>)` (if we hypothetically had access to the dynamically constructed class), whereas `isinstance(module, FSDPLinear)` seems syntactically valid.

Reply:
Oh it's mainly stylistic difference and I was only thinking about sth like `__str__` to nicely show it's a FSDP wrapped module, but the `isinstance` check make sense we should just have a valid class name without `<>`